reactor-core/src/main/java/reactor/core/publisher/Flux.java

```diff
@@ -3497,11 +3497,43 @@ public final T blockLast(Duration timeout) {
 	public final <V, C extends Collection<? super V>> Flux<T> distinct(
 			Function<? super T, ? extends V> keySelector,
 			Supplier<C> distinctCollectionSupplier) {
+		return this.distinct(keySelector, distinctCollectionSupplier, Collection::add, Collection::clear);
```

Comment:
could store static variable in FluxDistinct

Reply:
I don't think method references benefit from storing in a static variable